pkg/naming/labels.go

```diff
@@ -26,12 +30,23 @@ func DatacenterLabels(c *scyllav1.ScyllaCluster) map[string]string {
 
 // RackLabels returns a map of label keys and values
 // for the given Rack.
-func RackLabels(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster) map[string]string {
+func RackLabels(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster) (map[string]string, error) {
 	recLabels := ScyllaLabels()
 	rackLabels := DatacenterLabels(c)
 	rackLabels[RackNameLabel] = r.Name
+	_, rackOrdinal, ok := slices.Find(c.Spec.Datacenter.Racks, func(rack scyllav1.RackSpec) bool {
+		if rack.Name == r.Name {
+			return true
+		}
 
-	return mergeLabels(rackLabels, recLabels)
+		return false
```

Comment:
nit: for a one shot like this maybe it would be enought to `return rack.Name == r.Name`?

Reply:
obviously, thanks for noticing that